### PR TITLE
[vtctld] Delete legacy dual purpose duration/int flag parsing

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This behavior was deprecated when we moved to `pflag` in #11350, and coming back to clean this up to make viper easier.

## Related Issue(s)

#11456 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by deepthi — COMMENTED** [review]
Tests need to be fixed. They are passing the flag as an int.

**Review by shlomi-noach — APPROVED** [review]

File: changelog/17.0/17.0.0/summary.md [hunk 1]
```diff
@@ -257,6 +259,14 @@ These stats are deprecated in v17.
 | `backup_duration_seconds` | `BackupDurationNanoseconds` |
 | `restore_duration_seconds` | `RestoreDurationNanoseconds` |
 
+### <a id="vtctld"/> Vtctld
+
+#### <a id="vtctld-deprecated-flags"/> Deprecated Flags
+
+The flag `schema_change_check_interval` used to accept either a Go duration value (e.g. `1m` or `30s`) or a bare integer, which was treated as seconds.
+This behavior was deprecated in v15.0.0 and has been removed.
```
Comment (deepthi) [inline changelog/17.0/17.0.0/summary.md:267]: I hope we announced the deprecation in v15

**Review by deepthi — APPROVED** [review]
Changes LGTM. Can you check whether the website command reference is up to date? If it is, then the website docs label can be removed.

File: changelog/17.0/17.0.0/summary.md (hunk 1 above)
Comment (ajm188) [reply changelog/17.0/17.0.0/summary.md:267]: the deprecation shipped in 15.0 and has been dutifully issuing depreciation warnings ever since

**Pull request merged**